src/views/List.jsx

```diff
@@ -1,15 +1,44 @@
 import { ListItem } from '../components';
+import { useState } from 'react';
 
 export function List({ data }) {
+	const [searchInput, setSearchInput] = useState('');
+
 	const listItemsToDisplay = data.map((item) => {
-		return <ListItem key={item.id} name={item.name} />;
+		return item.name?.toLowerCase().includes(searchInput.toLowerCase()) ? (
+			<ListItem key={item.id} name={item.name} />
+		) : null;
 	});
 
+	const resetDisplayList = (e) => {
+		e.preventDefault();
+		setSearchInput('');
+	};
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
+			<form>
+				<label htmlFor="filter">Filter by item name</label>
+				<br />
+				<input
+					type="text"
+					name="filter"
+					id="filter"
+					value={searchInput}
+					placeholder="Name"
+					onChange={(e) => {
+						setSearchInput(e.target.value);
+					}}
+				></input>
+				<button
+					name="clearInput"
+					onClick={(e) => {
+						resetDisplayList(e);
+					}}
+				>
+					X
+				</button>
```

Comment:
The fact that you didn't specify a `type` on the button element would make it use `type="submit"` 😱 and that's not what we want, instead, we want a button that doesn't submit anything.

Let's add the `type="button"` attribute before merging, please!

![CleanShot 2023-08-31 at 18 47 29@2x](https://github.com/the-collab-lab/tcl-64-smart-shopping-list/assets/464978/e42b8a8c-0e61-47ef-9c9d-e1a737610fae)

## MDN Docs
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button

Reply:
Thank you for this callout, I wasn't aware of this default behavior! I have added `type="button"` in the latest commits.